flink-connectors/flink-connector-kafka/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/KafkaConsumerThread.java

```diff
@@ -503,6 +506,17 @@ KafkaConsumer<byte[], byte[]> getConsumer(Properties kafkaProperties) {
         return new KafkaConsumer<>(kafkaProperties);
     }
 
+    private void retryOnWakeup(Runnable consumerCall, String description) {
+        try {
+            consumerCall.run();
+        } catch (WakeupException we) {
+            log.info(
+                    "Caught WakeupException while executing Kafka consumer call for {}. Will retry the consumer call.",
```

Comment:
Caught WakeupException while executing Kafka consumer call for {}. Will retry it once.